### PR TITLE
AZR-177: Azure Weka Essential sanity-testing.yaml workflow

### PR DESCRIPTION
**Pull request by DmytroWekaQa**
Working example: https://github.com/weka/terraform-azure-weka-essential-test/actions/runs/5459026902/jobs/9934628114

### CODE REVIEW

File: ci/deployment/test_sanity_weka_cluster.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import os
+import re
+import time
+
+from weka_rest_api_client.api import API
+from weka_rest_api_client.errors import NoAvailableHosts, BadGatewayError, RestClientError
+
+
+def parse_ips_from_tf_output():
```
Comment (assafgi) [inline ci/deployment/test_sanity_weka_cluster.py:9]: you should use `terraform output -json` then you can parse it load the output and parse it easily.

File: create_service_principal.tf [hunk 2]
```diff
@@ -0,0 +1,146 @@
+#provider "azurerm" {
```
Comment (assafgi) [inline create_service_principal.tf:1]: shouldn't be part of this PR

File: examples/no_existing_network/main.tf [hunk 3]
```diff
@@ -1,18 +1,19 @@
 provider "azurerm" {
-  subscription_id = "..."
+  subscription_id = var.subscription_id
```
Comment (assafgi) [inline examples/no_existing_network/main.tf:2]: you shouldn't change the examples

File: examples/no_existing_network/vars.auto.tfvars [hunk 4]
```diff
@@ -0,0 +1,5 @@
+prefix = "essential"
```
Comment (assafgi) [inline examples/no_existing_network/vars.auto.tfvars:1]: you shouldn't change examples, you can pass all the variables you need directly, no need for this file

File: variables.tf [hunk 5]
```diff
@@ -284,3 +284,14 @@ variable "mount_clients_dpdk" {
   default     = true
   description = "Mount weka clients in DPDK mode"
 }
+
+variable "subscription_id" {
+  type = string
+  description = "The subscription id for the deployment."
```
Comment (assafgi) [inline variables.tf:290]: you shouldn't add variables to our module just for testing

File: .github/workflows/sanity-testing.yaml [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+        run: cat ../../ci/deployment/additional_variables.tf >> variables.tf
+
+      - name: Terraform Init
+        run: terraform init
+
+      - name: Terraform Apply
+        run: terraform apply -var-file vars.auto.tfvars -var="prefix=$PREFIX" -var="rg_name=$RG_NAME" -var="cluster_name=$CLUSTER_NAME" -var="cluster_size=$INITIAL_CAPACITY" -var="get_weka_io_token=$GET_WEKA_IO_TOKEN" -var="client_id=$CLIENT_ID" -var="tenant_id=$TENANT_ID" -var="client_secret=$CLIENT_SECRET" -var="subscription_id=$SUBSCRIPTION_ID" -auto-approve
+
+      - name: Terraform Output
+        id: tfoutput
```
Comment (assafgi) [inline .github/workflows/sanity-testing.yaml:63]: maybe if you use `terrraform output -json` you will not encounter this issue

File: ci/deployment/test_sanity_weka_cluster.py (hunk 1 above)
Comment (DmytroWekaQa) [reply ci/deployment/test_sanity_weka_cluster.py:9]: I tried it - doesn't helpful too much, because anyway I need to pass this between steps somehow and Github actions produce errors like in https://github.com/weka/terraform-azure-weka-essential-test/actions/runs/5456520184/jobs/9929435616 on multiline strings - that's why I anyway need to cut it to single line to be able to stream into $GITHUB_OUTPUT

File: create_service_principal.tf (hunk 2 above)
Comment (DmytroWekaQa) [reply create_service_principal.tf:1]: I added this because we discussed doing that for future changes.  Would you like me to remove it?

File: variables.tf (hunk 5 above)
Comment (DmytroWekaQa) [reply variables.tf:290]: As I remember I was made to add this because of create_service_principal.tf addition.
If we will decide to remove create_service_principal.tf I will be able to cut this off also.

File: ci/deployment/test_sanity_weka_cluster.py (hunk 1 above)
Comment (assafgi) [reply ci/deployment/test_sanity_weka_cluster.py:9]: so more specific maybe:
```terraform output -json | python -c "import json;import sys; print(json.load(sys.stdin)['weka_deployment_output']['value']['backend_ips'])"```

File: variables.tf (hunk 5 above)
Comment (assafgi) [reply variables.tf:290]: Service principal creation isn't relevant here.
These are our root module variables and our tests shouldn't touch it.

File: create_service_principal.tf (hunk 2 above)
Comment (DmytroWekaQa) [reply create_service_principal.tf:1]: Removed

File: examples/no_existing_network/main.tf (hunk 3 above)
Comment (DmytroWekaQa) [reply examples/no_existing_network/main.tf:2]: Revert this change

File: examples/no_existing_network/vars.auto.tfvars (hunk 4 above)
Comment (DmytroWekaQa) [reply examples/no_existing_network/vars.auto.tfvars:1]: Removed file

File: variables.tf (hunk 5 above)
Comment (DmytroWekaQa) [reply variables.tf:290]: Revert this changes

**Comment by DmytroWekaQa** [comment]
Will be reopened after I'll fix all comments and re-test solution in forked repo
